From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 30 Mar 2009 11:30:29 -0400
Subject: [misc] add HP xw460c to bf sort pci list
Message-id: 20090330153028.29970.5698.sendpatchset@prarit.bos.redhat.com
O-Subject: [RHEL5 PATCH]: Add HP xw460c to bf sort pci list
Bugzilla: 490068
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>

Add HP ProLiant xw460c Blade Workstation to the breadth-first PCI sort list
so that the ethernet ports are enumerated properly.

Brew built and tested by HP.

Resolves BZ 490068.

diff --git a/arch/i386/pci/common.c b/arch/i386/pci/common.c
index f95f0ac..8373c7e 100644
--- a/arch/i386/pci/common.c
+++ b/arch/i386/pci/common.c
@@ -339,6 +339,15 @@ static struct dmi_system_id __devinitdata pciprobe_dmi_table[] = {
 			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant DL580 G5"),
 		},
 	},
+	{
+		.callback = set_bf_sort,
+		.ident = "HP ProLiant xw460c",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME,
+				  "ProLiant xw460c Blade Workstation"),
+		},
+	},
 	{}
 };
 
